# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsISocketTransport.idl
# Commit: 521c9e019678
# Full Hash: 521c9e019678f584590f63473416d40e03f33342
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 04:10:55
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r dfb39ca05529 -r 521c9e019678 netwerk/base/nsISocketTransport.idl
--- a/netwerk/base/nsISocketTransport.idl	Tue Mar 14 15:29:49 2023 +0200
+++ b/netwerk/base/nsISocketTransport.idl	Tue Mar 14 12:59:40 2023 +0000
@@ -5,6 +5,7 @@
 
 #include "nsITransport.idl"
 #include "nsIRequest.idl"
+#include "nsITRRSkipReason.idl"
 
 interface nsIInterfaceRequestor;
 interface nsINetAddr;
@@ -346,6 +347,16 @@
     bool resolvedByTRR();
 
     /**
+     * Returns the effectiveTRRMode used for the DNS resolution.
+     */
+    readonly attribute nsIRequest_TRRMode effectiveTRRMode;
+
+    /**
+     * Returns the TRR skip reason used for the DNS resolution.
+     */
+    readonly attribute nsITRRSkipReason_value trrSkipReason;
+
+    /**
      * Indicate whether this socket is created from a private window. If yes,
      * this socket will be closed when the last private window is closed.
      */